feat(ppo): support async ppo (#3410)
NVIDIA-NeMo/RL: .agents/contributor-skills/config-conventions/SKILL.md

**v1 (legacy, being migrated away):**

> - **`typing.TypedDict` — v1, legacy / not-yet-migrated user-facing config.** Most nested sub-configs (e.g. `GRPOConfig`, `RewardScalingConfig`, `AdvEstimatorConfig`) are still `TypedDict`. Continue to maintain them with the same defaults rules below until they are migrated to `BaseModel`. Use `typing.NotRequired` to mark optional attributes. **Do not add new `TypedDict`-based config classes.**

Reviewer: nit: not introduced in this PR. currently we updated many config things but forgot to update here, could you help also fix it?
```suggestion
- **`typing.TypedDict` — v1, legacy / not-yet-migrated user-facing config.** Some nested sub-configs are still `TypedDict`. Continue to maintain them with the same defaults rules below until they are migrated to `BaseModel`. Use `typing.NotRequired` to mark optional attributes. **Do not add new `TypedDict`-based config classes.**
```

Author: Agreed. I’ll apply the suggested update here. I also opened #3598 to track the broader config-conventions cleanup separately so it doesn’t block this PR.

Reviewer: fixed in 2fb2ff68f9a52120323d4e73d81bc1d99ae6ccad
